src/commonMain/kotlin/com/algolia/search/endpoint/EndpointSearchImpl.kt

```diff
@@ -35,20 +35,29 @@ internal class EndpointSearchImpl(
         return transport.request(HttpMethod.Post, CallType.Read, indexName.toPath("/query"), requestOptions, body)
     }
 
-    override tailrec suspend fun findFirstObject(
+    override suspend fun findFirstObject(
         match: (ResponseSearch.Hit) -> Boolean,
         query: Query,
         doNotPaginate: Boolean,
         requestOptions: RequestOptions?
+    ): ResponseHitWithPosition? {
+        return findObject(match, query, doNotPaginate, requestOptions)
```

Comment:
You don't need to inverse `doNotPaginate` to `! doNotPaginate` ?

Reply:
you make some good points 👍